src/ngraph/pass/core_fusion.cpp

```diff
@@ -73,3 +81,89 @@ void pass::CoreFusion::construct_relu()
+{
+    Shape shape{2, 2, 1, 1};
+    auto input = std::make_shared<pattern::op::Label>(element::f32, shape);
+    auto filters = std::make_shared<pattern::op::Label>(element::f32, shape);
+
+    auto pconv = std::make_shared<op::Convolution>(input,
+                                                   filters,
+                                                   Strides{1, 1},
+                                                   Strides{1, 1},
+                                                   CoordinateDiff{0, 0},
+                                                   CoordinateDiff{0, 0},
+                                                   Strides{1, 1});
+    auto mean_shape = Shape{2};
+    auto mean = std::make_shared<pattern::op::Label>(element::f32, mean_shape);
+    auto var_shape = Shape{2};
+    auto var = std::make_shared<pattern::op::Label>(element::f32, var_shape);
+    auto gamma_shape = Shape{2};
+    auto gamma = std::make_shared<pattern::op::Label>(element::f32, gamma_shape);
+    auto beta_shape = Shape{2};
+    auto beta = std::make_shared<pattern::op::Label>(element::f32, beta_shape);
+    double eps = 0.001;
+    auto shape_r = Shape{1, 2, 2, 2};
+    auto bn = std::make_shared<op::BatchNorm>(eps, gamma, beta, pconv, mean, var);
+
+    ngraph::pattern::graph_rewrite_callback callback = [input, filters, mean, var, gamma, beta](
+        pattern::Matcher& m) {
+        NGRAPH_DEBUG << "In callback for folded batch norm against node = "
+                     << m.get_match_root()->get_name();
+        auto pattern_map = m.get_pattern_map();
+
+        auto m_bn = std::dynamic_pointer_cast<op::BatchNorm>(m.get_match_root());
+        auto m_conv = std::dynamic_pointer_cast<op::Convolution>(m_bn->get_argument(2));
+
+        if (m_conv->get_users().size() > 1)
+        {
+            return false;
+        }
+
+        if (m_conv->get_element_type() != m_bn->get_element_type())
+        {
+            return false;
+        }
+
+        if (m_conv->get_shape().size() != 4)
+        {
+            return false;
+        }
+
+        // new weights = old weights * gamma / sqrt(variance + epsilon)
+        // new biases = -mean * gamma / sqrt(variance + epsilon) + beta
+
+        auto bn_eps = op::Constant::create(element::f32, Shape{}, {m_bn->get_eps_value()});
+        auto var_eps = std::make_shared<op::Add>(
+            pattern_map[var],
+            std::make_shared<op::Broadcast>(bn_eps, pattern_map[var]->get_shape(), AxisSet{0}));
+        auto sqrt_var_eps = std::make_shared<op::Sqrt>(var_eps);
+
+        auto mean_gamma = std::make_shared<op::Multiply>(pattern_map[mean], pattern_map[gamma]);
+        auto new_biases = std::make_shared<op::Subtract>(
+            pattern_map[beta], std::make_shared<op::Divide>(mean_gamma, sqrt_var_eps));
+        auto weight_scaling = std::make_shared<op::Divide>(pattern_map[gamma], sqrt_var_eps);
+        auto new_filters = std::make_shared<op::Multiply>(
```

Comment:
please either rename `new_filters` to `new_weights` or vice versa so yr comment matches the logic